contracts/badger-ren/BadgerRenAdapter.sol

```diff
@@ -0,0 +1,272 @@
+        address payable _destination,
+        // darknode args
+        uint256 _amount,
+        bytes32 _nHash,
+        bytes calldata _sig
+    ) external nonReentrant {
+        // Mint renBTC tokens
+        bytes32 pHash = keccak256(abi.encode(_destination));
+        uint256 _mintAmount = registry.getGatewayBySymbol("BTC").mint(pHash, _amount, _nHash, _sig);
+        uint256 _fee = _processFee(renBTC, _mintAmount, mintFeeBps);
+
+        emit MintRenBTC(_mintAmount, _fee);
+
+        renBTC.safeTransfer(_destination, _mintAmount.sub(_fee));
+    }
+
+    function burnRenBTC(bytes calldata _btcDestination, uint256 _amount) external nonReentrant {
+        require(renBTC.balanceOf(address(msg.sender)) >= _amount);
+        uint256 _startBalance = renBTC.balanceOf(address(this));
+        renBTC.safeTransferFrom(msg.sender, address(this), _amount);
+        uint256 _endBalance = renBTC.balanceOf(address(this));
+
+        uint256 _burnAmount = _endBalance.sub(_startBalance);
+        uint256 _fee = _processFee(renBTC, _burnAmount, burnFeeBps);
+
+        emit BurnRenBTC(_burnAmount, _fee);
+
+        uint256 burnAmount = registry.getGatewayBySymbol("BTC").burn(_btcDestination, _burnAmount.sub(_fee));
+    }
+
+    function mintWBTC(
+        uint256 _slippage,
+        address payable _destination,
+        uint256 _amount,
+        bytes32 _nHash,
+        bytes calldata _sig
+    ) external nonReentrant {
+        // Mint renBTC tokens
+        bytes32 pHash = keccak256(abi.encode(_slippage, _destination));
+        uint256 _mintAmount = registry.getGatewayBySymbol("BTC").mint(pHash, _amount, _nHash, _sig);
+        _swapRenBTCForWBTC(_slippage, _destination, _mintAmount);
+    }
+
+    function _swapRenBTCForWBTC(
+        uint256 _slippage,
+        address payable _destination,
+        uint256 _mintAmount
+    ) internal {
+        // Get price
+        uint256 dy = exchange.get_dy(0, 1, _mintAmount);
+        _slippage = uint256(1e4).sub(_slippage);
+        uint256 min_dy = dy.mul(_slippage).div(1e4);
+
+        uint256 _startBalance = wBTC.balanceOf(address(this));
+        try exchange.exchange(0, 1, _mintAmount, min_dy)  {
+            uint256 _endBalance = wBTC.balanceOf(address(this));
+            uint256 _wbtcAmount = _endBalance.sub(_startBalance);
+            uint256 _fee = _processFee(wBTC, _wbtcAmount, mintFeeBps);
+            emit MintWBTC(_mintAmount, _wbtcAmount, _fee);
+
+            // Send converted wBTC to user.
+            require(wBTC.transfer(_destination, _wbtcAmount.sub(_fee)));
+        } catch Error(string memory _error) {
+            emit ExchangeWBTCStringError(_error);
+        } catch (bytes memory _error) {
+            emit ExchangeWBTCBytesError(_error);
+        }
+        _fallbackTransferRenBTC(_destination, _mintAmount);
+    }
+
+    function burnWBTC(
+        bytes calldata _btcDestination,
+        uint256 _amount,
+        uint256 _minAmount
+    ) external nonReentrant {
```

Comment:
Consider asserting `_minAmount` is not going to cause too much damage to users swap, I appreciate this might be ambiguous ask but we should try and pretend user value over the swap